### PR TITLE
Fix part of #174: Add test to ensure that bot doesn't gets triggred with "@{{username}} PTAL" comment

### PR DESCRIPTION
**Pull request by mridul-netizen**
## Explanation

Fix part of #174: Add test to ensure that bot doesn't gets triggred with "@{{username}} PTAL" comment
- [x] I have successfully deployed my own instance of Oppiabot.
  - You can find instructions for doing this [here](https://github.com/oppia/oppiabot/wiki/Deploying-your-own-instance-of-the-oppiabot).
- [x] I have manually tested all the changes made in this PR following the [manual tests matrix](https://github.com/oppia/oppiabot/wiki/Manual-Tests-Matrix).

### CODE REVIEW

**Comment by mridul-netizen** [comment]
@jameesjohn PTAL!

**Review by jameesjohn — CHANGES_REQUESTED** [review]
Thanks, @mridul-netizen.

I'm not sure the test block actually tests that oppiabot doesn't get triggered with the comment pattern.
One way to test this would be to make update the comment payload for this test to include the required comment and assert that the bot doesn't get triggered.

**Comment by mridul-netizen** [comment]
@jameesjohn PTAL thanks!

File: spec/checkPullRequestReviewSpec.js [hunk 1]
```diff
@@ -1522,5 +1522,23 @@ describe('Pull Request Review Module', () => {
         });
       }
     );
+    describe(
+      'when author asks some question/comment and asks reviewer to ptal',
+      () => {
+        const initialCommentBody = commentPayloadData.payload.comment.body;
+        beforeAll(() => {
+          commentPayloadData.payload.comment.body =
+          'Hi I have a doubt @reviewer PTAL';
```
Comment (jameesjohn) [inline spec/checkPullRequestReviewSpec.js:1531]: Hi @DubeySandeep, does this comment satisfy the use case?

**Review by jameesjohn — APPROVED** [review]
LGTM!
Just requires some confirmation from @DubeySandeep.

File: spec/checkPullRequestReviewSpec.js [hunk 2]
```diff
@@ -1522,5 +1522,23 @@ describe('Pull Request Review Module', () => {
...
+        });
+        beforeEach(async () => {
+          await robot.receive(commentPayloadData);
+        });
+        it('should not comment on pull request', () => {
+          expect(github.issues.createComment).not.toHaveBeenCalled();
```
Comment (DubeySandeep) [inline spec/checkPullRequestReviewSpec.js:1537]: I was expecting `Hi I have a doubt @reviewer PTAL` to trigger bot to assign @reviewer. [@jameesjohn Can you confirm this and check how this test passed here?]

And through issue #174 I meant a message like  `{{Question/comment}} @{{reviewer_username}} PTAL` should not trigger the bot. [Note: {{}} here are not meant for variable values i.e, not for interpolation]

**Review by DubeySandeep — CHANGES_REQUESTED** [review]

File: spec/checkPullRequestReviewSpec.js [hunk 3]
```diff
@@ -1522,5 +1522,23 @@ describe('Pull Request Review Module', () => {
         });
       }
     );
+    describe(
```
[the remark this answers is not in the document]
Comment (mridul-netizen) [reply spec/checkPullRequestReviewSpec.js:1526]: Done!

File: spec/checkPullRequestReviewSpec.js [hunk 4]
```diff
@@ -1522,5 +1522,23 @@ describe('Pull Request Review Module', () => {
...
+      () => {
+        const initialCommentBody = commentPayloadData.payload.comment.body;
+        beforeAll(() => {
+          commentPayloadData.payload.comment.body =
+          'Hi I have a doubt @reviewer PTAL';
+        });
+        beforeEach(async () => {
+          await robot.receive(commentPayloadData);
+        });
+        it('should not comment on pull request', () => {
```
[the remark this answers is not in the document]
Comment (mridul-netizen) [reply spec/checkPullRequestReviewSpec.js:1538]: Done!

File: spec/checkPullRequestReviewSpec.js (hunk 2 above)
Comment (mridul-netizen) [reply spec/checkPullRequestReviewSpec.js:1539]: @DubeySandeep I updated the above pattern it passed with this one too

File: spec/checkPullRequestReviewSpec.js [hunk 5]
```diff
@@ -1522,5 +1522,26 @@ describe('Pull Request Review Module', () => {
...
+        const initialCommentBody = commentPayloadData.payload.comment.body;
+        beforeAll(() => {
+          commentPayloadData.payload.comment.body =
+          '{{Question/comment}} @{{reviewer_username}} PTAL';
+        });
+        beforeEach(async () => {
+          await robot.receive(commentPayloadData);
+        });
+
+        it('should not comment on pull request', () => {
```
Comment (DubeySandeep) [inline spec/checkPullRequestReviewSpec.js:1538]: `should not trigger on comment with PTAL template`

File: spec/checkPullRequestReviewSpec.js [hunk 6]
```diff
@@ -1522,5 +1522,26 @@ describe('Pull Request Review Module', () => {
...
+          expect(github.issues.createComment).not.toHaveBeenCalled();
```
Comment (DubeySandeep) [inline spec/checkPullRequestReviewSpec.js:1539]: I'm finding it difficult to understand what's been tested here! Can you please help me with the link to the test which validates oppiaBot comments when someone leaves a PTAL comment?

Comment (mridul-netizen) [reply spec/checkPullRequestReviewSpec.js:1539]: https://github.com/oppia/oppia/pull/12132#issuecomment-803470984 @DubeySandeep does this one satisfy the test?

**Comment by mridul-netizen** [comment]
@jameesjohn @DubeySandeep PTAL

File: spec/checkPullRequestReviewSpec.js (hunk 2 above)
Comment (jameesjohn) [reply spec/checkPullRequestReviewSpec.js:1539]: @mridul-netizen, I think @DubeySandeep meant a message following that format. Not the exact text.

**Review by jameesjohn — CHANGES_REQUESTED** [review]
Following this comment, there should be some changes in the functionality for responding to the user's comment.

File: spec/checkPullRequestReviewSpec.js (hunk 2 above)
Comment (mridul-netizen) [reply spec/checkPullRequestReviewSpec.js:1539]: Can you please suggest changes needed to be made?

**Comment by Naman-1234** [comment]
@vojtechjelinek  and @jameesjohn Sir Author is inactive for a long time and this issue is of Medium Priority. This issue seems  really interesting to me. I would like to work on this if you grant. Could I be assigned on this one. Thanks.

**Comment by mridul-netizen** [comment]
Hi @Naman-1234 I'm not sure about some of the last comments that reviewers made here.If you could help me figure that out or suggest some code changes that'd be great!

**Comment by Naman-1234** [comment]
@mridul-netizen  Sir is asking for some response to the user's comment. I think for more reference you should mention sir to brief you about that.

File: spec/checkPullRequestReviewSpec.js [hunk 7]
```diff
@@ -1522,5 +1522,26 @@ describe('Pull Request Review Module', () => {
         });
       }
     );
+
+    describe(
+      'when author asks some question/comment and asks reviewer to ptal',
```
[the remark this answers is not in the document]
Comment (mridul-netizen) [reply spec/checkPullRequestReviewSpec.js]: done

**Comment by mridul-netizen** [comment]
@DubeySandeep PTAL

File: spec/checkPullRequestReviewSpec.js [hunk 8]
```diff
@@ -1522,5 +1522,26 @@ describe('Pull Request Review Module', () => {
         });
       }
     );
+
+    describe(
+      'when PTAL template found in the comment',
+      () => {
```
Comment (DubeySandeep) [inline spec/checkPullRequestReviewSpec.js:1528]:
```suggestion
    describe('when PTAL template found in the comment', () => {
```

File: spec/checkPullRequestReviewSpec.js [hunk 9]
```diff
@@ -1522,5 +1522,26 @@ describe('Pull Request Review Module', () => {
...
+        const initialCommentBody = commentPayloadData.payload.comment.body;
+        beforeAll(() => {
+          commentPayloadData.payload.comment.body =
+          '{{Question/comment}} @{{reviewer_username}} PTAL';
```
Comment (DubeySandeep) [inline spec/checkPullRequestReviewSpec.js:1532]:
```suggestion
          commentPayloadData.payload.comment.body = (
          '{{Question/comment}} @{{reviewer_username}} PTAL');
```
Always break after parenthesis.

**Review by DubeySandeep — APPROVED** [review]
LGTM, left a few minor nit comments! [Approved!]

Feel free to merge this PR once the nit comments are resolved!

File: spec/checkPullRequestReviewSpec.js (hunk 8 above)
Comment (mridul-netizen) [reply spec/checkPullRequestReviewSpec.js]: done

File: spec/checkPullRequestReviewSpec.js (hunk 9 above)
Comment (mridul-netizen) [reply spec/checkPullRequestReviewSpec.js]: done

**Comment by mridul-netizen** [comment]
@jameesjohn PTAL

**Review by jameesjohn — APPROVED** [review]

**Pull request merged**